nomad/server.go

```diff
@@ -1151,18 +1151,13 @@ func (s *Server) setupRpcServer(server *rpc.Server, ctx *RPCContext) {
 	if s.staticEndpoints.Status == nil {
 		// Initialize the list just once
 		s.staticEndpoints.ACL = &ACL{srv: s, logger: s.logger.Named("acl")}
-		s.staticEndpoints.Alloc = &Alloc{srv: s, logger: s.logger.Named("alloc")}
-		s.staticEndpoints.Eval = &Eval{srv: s, logger: s.logger.Named("eval")}
 		s.staticEndpoints.Job = NewJobEndpoints(s)
-		s.staticEndpoints.Node = &Node{srv: s, logger: s.logger.Named("client")} // Add but don't register
 		s.staticEndpoints.CSIVolume = &CSIVolume{srv: s, logger: s.logger.Named("csi_volume")}
 		s.staticEndpoints.CSIPlugin = &CSIPlugin{srv: s, logger: s.logger.Named("csi_plugin")}
-		s.staticEndpoints.Deployment = &Deployment{srv: s, logger: s.logger.Named("deployment")}
 		s.staticEndpoints.Operator = &Operator{srv: s, logger: s.logger.Named("operator")}
 		s.staticEndpoints.Operator.register()
 
 		s.staticEndpoints.Periodic = &Periodic{srv: s, logger: s.logger.Named("periodic")}
-		s.staticEndpoints.Plan = &Plan{srv: s, logger: s.logger.Named("plan")}
```

Comment:
I don't know if there's any performance impact in moving these endpoints out of `staticEndpoits` since the connection won't be re-used.

Reply:
Short answer: the security correctness is worth the negligible performance cost.

Long answer because I couldn't help but think about it: Let's do the math! Your change creates new structs on a per-connection basis instead of once globally. `srv` and `ctx` are both just pointers to shared structs, so that's at worst 16 bytes of overhead per connection.

`logger` is much bigger. Since it's an interface that's 16 bytes for the `(Type, Pointer)` tuple. Luckily `Named()` just creates a shallow of most of the 128 bytes worth of fields on logger. The exceptions being: `name` (16 bytes max I think) and `level` (4 bytes).

So logger adds something like 16 + 128 + 16 + 4 = 164 bytes per connection for a total of 180 bytes per connection.

 If somebody manages to make 50k connections to a Nomad server, that's a whopping 9MB of extra memory used _total._ Even if I'm off a couple orders of magnitude I think it's not worth worrying about.

**Memory wise we're fine.** :sweat_smile:

Memory allocations themselves cost something as well, and you're adding something like 6 per connection.

If that's enough to cause a noticeable difference we have been so incredibly efficient I think we can afford the extra cost. :sweat_smile: 

If we really cared we could create static stubs for these RPC methods, do the name check in setup, and only attach the real methods when a server is connecting. Absolutely not worth worrying about.

Reply:
Nice, thanks for the detailed analysis!